### ISSUE TITLE
Bug Report: Crash on dreath during thorns erect/normal

### ISSUE THREAD

**Issue by ChickenSwimmer2020** [FunkinCrew/Funkin#2801]
## Describe the bug
game crashes due to an issue with FlxTrail.hx

## To Reproduce
reset on thorns and restart
## Expected behavior
song restart

## Screenshots/Video
![image_2024-06-13_231954802](https://github.com/FunkinCrew/Funkin/assets/105177647/377c5ae2-7482-4c90-81bd-c0d223a875bf)

## Desktop
 - OS:
 Windows 11
 - Version:
0.4.1

**Comment by Hundrec** [comment]
I'm resetting on Thorns fine after starting in both Freeplay and the Chart Editor.
Did you use the chart editor before playing Thorns?

**Comment by ChickenSwimmer2020** [comment]
no, i didn't touch the chart editor at all

**Comment by GoombarioFan** [comment]
I just tested this and got a crash myself. After removing all my mods and testing each of them one by one, I found that it was somehow the Cassette Girl port by MongMong causing the crash to happen.

**Comment by Hundrec** [comment]
> I just tested this and got a crash myself. After removing all my mods and testing each of them one by one, I found that it was somehow the Cassette Girl port by MongMong causing the crash to happen.

Interesting! Did you also get the same error with Thorns?

Question for the poster:
Did you have anything in your mods folder?

**Comment by GoombarioFan** [comment]
> Interesting! Did you also get the same error with Thorns?
Yes, I got it only on Thorns. I tested it on a couple of more songs, including Senpai and Roses but did not crash. Afterwards, I went on the Chart Editor and changed a random song's opponent to Spirit, reset and caused a crash. So somehow, it's related to Spirit specifically.

**Comment by Hundrec** [comment]
Got it

If it's just a problem with Spirit, then I'm pretty sure FlxTrail.hx handles Spirit's crossfading effect when he moves around

Weird that it only happens in a few cases!

**Comment by vincethestick** [comment]
> I just tested this and got a crash myself. After removing all my mods and testing each of them one by one, I found that it was somehow the Cassette Girl port by VinceTheStick causing the crash to happen.

After some testing I found that the change character event scripts in the port was the culprit; after deleting the 'changecharactereventv2' folder thorns erect doesn't crash anymore. I also tested with a port of Zardy and Furnace Whitty which also had their own change character script files and caused thorns to crash. Idk why it's being funny with spirit specifically.

**Comment by Hundrec** [comment]
Has anyone experienced a crash on Thorns in 0.5.1?
